### ISSUE TITLE
rust-1.23 fails at minicargo

### ISSUE THREAD

**Issue by Gottox** [thepowersgang/mrustc#53]
I (naively) tried to build rust 1.23.0 with mrustc.

It fails while running minicargo. I straced it, here's the output:

```
# strace tools/bin/minicargo rustc-1.23.0-src/src/libstd --script-overrides script-overrides/stable-1.23.0-linux/ --output-dir output/ -j 1

execve("tools/bin/minicargo", ["tools/bin/minicargo", "rustc-1.23.0-src/src/libstd", "--script-overrides", "script-overrides/stable-1.23.0-l"..., "--output-dir", "output/", "-j", "1"], 0x7ffe5ecfbca8 /* 61 vars */) = 0
brk(NULL)                               = 0x5610c8455000
access("/etc/ld.so.preload", R_OK)      = -1 ENOENT (No such file or directory)
openat(AT_FDCWD, "/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
fstat(3, {st_mode=S_IFREG|0644, st_size=145334, ...}) = 0
mmap(NULL, 145334, PROT_READ, MAP_PRIVATE, 3, 0) = 0x7f8911ed8000
close(3)                                = 0
openat(AT_FDCWD, "/usr/lib/libpthread.so.0", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\0b\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=2348680, ...}) = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f8911ed6000
mmap(NULL, 2221128, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f8911ab9000
mprotect(0x7f8911ad3000, 2093056, PROT_NONE) = 0
mmap(0x7f8911cd2000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x19000) = 0x7f8911cd2000
mmap(0x7f8911cd4000, 13384, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f8911cd4000
close(3)                                = 0
openat(AT_FDCWD, "/usr/lib/libstdc++.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\20+\t\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=2089240, ...}) = 0
mmap(NULL, 4197440, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f89116b8000
mprotect(0x7f89118a6000, 2093056, PROT_NONE) = 0
mmap(0x7f8911aa5000, 69632, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1ed000) = 0x7f8911aa5000
mmap(0x7f8911ab6000, 11328, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f8911ab6000
close(3)                                = 0
openat(AT_FDCWD, "/usr/lib/libm.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0000x\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=5205464, ...}) = 0
mmap(NULL, 3457800, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f891136b000
mprotect(0x7f89114b7000, 2093056, PROT_NONE) = 0
mmap(0x7f89116b6000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x14b000) = 0x7f89116b6000
close(3)                                = 0
openat(AT_FDCWD, "/usr/lib/libgcc_s.so.1", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\0\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0\300*\0\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0644, st_size=92576, ...}) = 0
mmap(NULL, 2188336, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f8911154000
mprotect(0x7f891116a000, 2093056, PROT_NONE) = 0
mmap(0x7f8911369000, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x15000) = 0x7f8911369000
close(3)                                = 0
openat(AT_FDCWD, "/usr/lib/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
read(3, "\177ELF\2\1\1\3\0\0\0\0\0\0\0\0\3\0>\0\1\0\0\0@\21\2\0\0\0\0\0"..., 832) = 832
fstat(3, {st_mode=S_IFREG|0755, st_size=16471224, ...}) = 0
mmap(NULL, 3885664, PROT_READ|PROT_EXEC, MAP_PRIVATE|MAP_DENYWRITE, 3, 0) = 0x7f8910d9f000
mprotect(0x7f8910f4a000, 2097152, PROT_NONE) = 0
mmap(0x7f891114a000, 24576, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_DENYWRITE, 3, 0x1ab000) = 0x7f891114a000
mmap(0x7f8911150000, 14944, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x7f8911150000
close(3)                                = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f8911ed4000
mmap(NULL, 12288, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f8911ed1000
arch_prctl(ARCH_SET_FS, 0x7f8911ed1740) = 0
mprotect(0x7f891114a000, 16384, PROT_READ) = 0
mprotect(0x7f8911369000, 4096, PROT_READ) = 0
mprotect(0x7f89116b6000, 4096, PROT_READ) = 0
mprotect(0x7f8911aa5000, 53248, PROT_READ) = 0
mprotect(0x7f8911cd2000, 4096, PROT_READ) = 0
mprotect(0x5610c6c4b000, 4096, PROT_READ) = 0
mprotect(0x7f8911efc000, 4096, PROT_READ) = 0
munmap(0x7f8911ed8000, 145334)          = 0
set_tid_address(0x7f8911ed1a10)         = 25856
set_robust_list(0x7f8911ed1a20, 24)     = 0
rt_sigaction(SIGRTMIN, {sa_handler=0x7f8911abec70, sa_mask=[], sa_flags=SA_RESTORER|SA_SIGINFO, sa_restorer=0x7f8911acb7d0}, NULL, 8) = 0
rt_sigaction(SIGRT_1, {sa_handler=0x7f8911abed00, sa_mask=[], sa_flags=SA_RESTORER|SA_RESTART|SA_SIGINFO, sa_restorer=0x7f8911acb7d0}, NULL, 8) = 0
rt_sigprocmask(SIG_UNBLOCK, [RTMIN RT_1], NULL, 8) = 0
prlimit64(0, RLIMIT_STACK, NULL, {rlim_cur=8192*1024, rlim_max=RLIM64_INFINITY}) = 0
brk(NULL)                               = 0x5610c8455000
brk(0x5610c8476000)                     = 0x5610c8476000
futex(0x7f8911ab6c9c, FUTEX_WAKE_PRIVATE, 2147483647) = 0
futex(0x7f8911ab6ca8, FUTEX_WAKE_PRIVATE, 2147483647) = 0
openat(AT_FDCWD, "rustc-1.23.0-src/src/libstd/Cargo.toml", O_RDONLY) = 3
read(3, "[package]\nauthors = [\"The Rust P"..., 8191) = 1489
read(3, "", 8191)                       = 0
openat(AT_FDCWD, "rustc-1.23.0-src/src/libstd/build.rs", O_RDONLY) = 4
close(4)                                = 0
close(3)                                = 0
openat(AT_FDCWD, "rustc-1.23.0-src/src/libstd/../liballoc/Cargo.toml", O_RDONLY) = 3
close(3)                                = 0
openat(AT_FDCWD, "rustc-1.23.0-src/src/liballoc/Cargo.toml", O_RDONLY) = 3
read(3, "[package]\nauthors = [\"The Rust P"..., 8191) = 393
read(3, "", 8191)                       = 0
close(3)                                = 0
openat(AT_FDCWD, "rustc-1.23.0-src/src/liballoc/../libcore/Cargo.toml", O_RDONLY) = 3
close(3)                                = 0
openat(AT_FDCWD, "rustc-1.23.0-src/src/libcore/Cargo.toml", O_RDONLY) = 3
read(3, "[package]\nauthors = [\"The Rust P"..., 8191) = 277
read(3, "", 8191)                       = 0
close(3)                                = 0
openat(AT_FDCWD, "rustc-1.23.0-src/src/liballoc/../libstd_unicode/Cargo.toml", O_RDONLY) = 3
close(3)                                = 0
openat(AT_FDCWD, "rustc-1.23.0-src/src/libstd_unicode/Cargo.toml", O_RDONLY) = 3
read(3, "[package]\nauthors = [\"The Rust P"..., 8191) = 268
read(3, "", 8191)                       = 0
close(3)                                = 0
openat(AT_FDCWD, "rustc-1.23.0-src/src/libstd_unicode/../libcore/Cargo.toml", O_RDONLY) = 3
close(3)                                = 0
openat(AT_FDCWD, "rustc-1.23.0-src/src/libstd/../liballoc_system/Cargo.toml", O_RDONLY) = 3
close(3)                                = 0
openat(AT_FDCWD, "rustc-1.23.0-src/src/liballoc_system/Cargo.toml", O_RDONLY) = 3
read(3, "[package]\nauthors = [\"The Rust P"..., 8191) = 474
rt_sigprocmask(SIG_UNBLOCK, [ABRT], NULL, 8) = 0
rt_sigprocmask(SIG_BLOCK, ~[RTMIN RT_1], [], 8) = 0
getpid()                                = 25856
gettid()                                = 25856
tgkill(25856, 25856, SIGABRT)           = 0
rt_sigprocmask(SIG_SETMASK, [], NULL, 8) = 0
--- SIGABRT {si_signo=SIGABRT, si_code=SI_TKILL, si_pid=25856, si_uid=1000} ---
+++ killed by SIGABRT +++
[1]    25854 abort      strace tools/bin/minicargo rustc-1.23.0-src/src/libstd --script-overrides    
```
`script-overrides/stable-1.23.0-linux` is a symlink to `script-overrides/stable-1.19.0-linux`

I guess it's due some parsing issues of the Cargo.toml.

**Comment by thepowersgang** [comment]
This is because of an unsupported target spec in the Cargo.toml for liballoc_system. (and incorrect logging of the error).
Note - The only version that's really supported is 1.19.0, because each version changes some of the internals of libcore/libstd in potentially incompatible ways.

**Comment by thepowersgang** [comment]
Parse error fixed in 9e864a5, but as said above, 1.23 fails to compile due to added features.

**Comment by Gottox** [comment]
Understood, thank you :)
